spec/rubocop/cop/root_cops/factory_file_name_spec.rb

```diff
@@ -0,0 +1,80 @@
+require_relative "../../../spec_helper.rb"
+
+RSpec.describe RuboCop::Cop::RootCops::FactoryFileName do
+  subject(:cop) { described_class.new }
+
+  before do
+    allow(File).to receive(:basename).and_return("factory_name")
```

Comment:
Does this only check factories named "factory_name"? Maybe add two contexts: `when the factory name has an underscore` and `when the factory name does not have an underscore`.

Reply:
Should I repeat all the tests for when the factory name does not have an underscore?

Reply:
I would.

Reply:
Sounds good thank you!